docker/Dockerfile.native

```diff
@@ -0,0 +1,31 @@
+# This dockerfile is used when the target matches the images platform in `build-docker-image`
+FROM ubuntu:20.04
+ARG DEBIAN_FRONTEND=noninteractive
+
+COPY common.sh lib.sh /
+RUN /common.sh
+
+COPY cmake.sh /
+RUN /cmake.sh
+
+COPY xargo.sh /
+RUN /xargo.sh
+
+ARG TARGETARCH
```

Comment:
This is done after the shared code, which will be nice when creating a base image but also for cached layers even before then.